avionics/common/include/state_events.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "HAL/time.h"
+#include "radio.h"
+#include "state_interface.h"
+
+namespace State {
+    const std::unordered_map<StateId, EventId> STATE_CHANGE_EVENTS = {
+        {StateId::ARMED, EventId::ARMED},
+        {StateId::STANDBY, EventId::DISARMED},
+        {StateId::POWERED_ASCENT, EventId::LAUNCH},
+        {StateId::MACH_LOCK, EventId::MACH_LOCK_ENTER},
+        {StateId::ASCENT_TO_APOGEE, EventId::MACH_LOCK_EXIT},
+        {StateId::PRESSURE_DELAY, EventId::APOGEE},
+        {StateId::DROGUE_DESCENT, EventId::DROGUE_DEPLOY},
+        {StateId::MAIN_DESCENT, EventId::MAIN_DEPLOY},
+        {StateId::LANDED, EventId::LANDED},
+        {StateId::WINTER_CONTINGENCY, EventId::ABORTED}
+    };
```

Comment:
I'm okay with this.

But just as a comment, I haven't used an unordered_map before, but I am curious what the lookup overhead is and if the compiler is able to optimize any of it out realizing that its const and static. Also wondering if this still uses dynamic memory. Curious to hear if @jesseli2002 has anything to say / opinions on this.

Reply:
- std::unordered_map uses dynamic memory but it'd be allocated once on startup so it's not as bad
- if you really wanted to optimize lookup, easiest thing to do would be instead use an array - those enums are constant.

Reply:
Function containing a switch statement marked as inline might be well optimized. Only downside I can see is the formatting isn't very nice. I wonder if anybody has a macro library to abstract that.

Reply:
switch statement like that would could be optimized to a jump into the switch block based on the actual integer value, or if the compiler is smarter an array lookup - so again you may as well use an array

Reply:
With the switch the "index"/mapping is explicit. If you use an array the mapping is implied by the order and isn't robust to people adding enum values or changing their order.

Reply:
to be fair adding enum values doesn't get solved by using switch statements either, since you still need to remember to update the switch statement. 
For that matter, it's arguably easier to test some level of validity with an array, since if you have a fixed "NUM" fake enum value at the end, you can iterate over all possible enum values in a test and check that they exist in the array